src/bernsteinyang.rs

```diff
@@ -0,0 +1,351 @@
+        if value.is_negative() {
+            value = value + &self.modulus;
+        }
+
+        if negate {
+            value = -value;
+        }
+
+        if value.is_negative() {
+            value = value + &self.modulus;
+        }
+
+        value
+    }
+
+    /// Returns a big unsigned integer as an array of O-bit chunks, which is equal modulo
+    /// 2 ^ (O * S) to the input big unsigned integer stored as an array of I-bit chunks.
+    /// The ordering of the chunks in these arrays is little-endian
+    const fn convert<const I:usize, const O:usize, const S:usize>(input: &[u64]) -> [u64; S] {
+        const fn min(a: usize, b: usize) -> usize { if a > b { b } else { a } }
+        let (total, mut output, mut bits) = (min(input.len() * I, S * O), [0; S], 0);
+          
+        while bits < total {
+            let (i, o) = (bits % I, bits % O);
+            output[bits / O] |= (input[bits / I] >> i) << o;
+            bits += min(I - i, O - o);
+        }
+        
+        let mask = u64::MAX >> (64 - O);
+        let mut filled = total / O + if total % O > 0 { 1 } else { 0 };
+    
+        while filled > 0 {
+            filled -= 1;
+            output[filled] &= mask;
+        }
+        
+        output
+    }
+
+    /// Returns the multiplicative inverse of the argument modulo 2^B. The implementation is based
+    /// on the Hurchalla's method for computing the multiplicative inverse modulo a power of two
+    const fn inv(value: u64) -> i64 {
+        let x = value.wrapping_mul(3) ^ 2;
+        let y = 1u64.wrapping_sub(x.wrapping_mul(value));
+        let (x, y) = (x.wrapping_mul(y.wrapping_add(1)), y.wrapping_mul(y));
+        let (x, y) = (x.wrapping_mul(y.wrapping_add(1)), y.wrapping_mul(y));
+        let (x, y) = (x.wrapping_mul(y.wrapping_add(1)), y.wrapping_mul(y));
+        (x.wrapping_mul(y.wrapping_add(1)) & ChunkInt::<B,L>::MASK) as i64
+    }
+
+    /// Creates the inverter for specified modulus and adjusting parameter
+    pub const fn new(modulus: &[u64], adjuster: &[u64]) -> Self {
+        Self {
+            modulus: ChunkInt::<B, L>(Self::convert::<64, B, L>(modulus)), 
+            adjuster: ChunkInt::<B, L>(Self::convert::<64, B, L>(adjuster)),
+            inverse: Self::inv(modulus[0])
+        }
+    }
+    
+    /// Returns either the adjusted modular multiplicative inverse for the argument or None 
+    /// depending on invertibility of the argument, i.e. its coprimality with the modulus
+    pub fn invert<const S:usize>(&self, value: &[u64]) -> Option<[u64; S]> {
+        let (mut d, mut e) = (ChunkInt::ZERO, self.adjuster.clone());
+        let mut g = ChunkInt::<B, L>(Self::convert::<64, B, L>(value)); 
+        let (mut delta, mut f) = (1, self.modulus.clone());
+        let mut matrix;
+        while g != ChunkInt::ZERO {
+            (delta, matrix) = Self::step(&f, &g, delta);
+            (f, g) = Self::fg(f, g, matrix);
```

Comment:
I thought about this during planing the implementation, but decided not to do this, because:
- It will require some extra time to determine the number of unused limbs after each arithmetic operation;
- Only the f,g-update can benefit from it;
- For the two's complement code working with fixed-length register-like structure is much more convenient.
For much larger modulus this optimization makes sense, but I am not sure that it is needed here.